frontend/src/hooks/store/useToastState.ts

```diff
@@ -0,0 +1,26 @@
+import { create } from 'zustand';
+
+interface ToastState {
+  isOpen: boolean;
+  text: string;
+  isSuccess: boolean;
+}
+
+interface ToastAction {
+  close: VoidFunction;
+
+  onSuccess: (text: string) => void;
+  onFailure: (text: string) => void;
+}
+
+const useToastState = create<ToastState & ToastAction>(set => ({
+  isOpen: false,
+  text: '',
+  isSuccess: false,
+
+  close: () => set({ isOpen: false }),
+  onSuccess: (text: string) => set(() => ({ isSuccess: true, text, isOpen: true })),
+  onFailure: (text: string) => set({ isSuccess: false, text, isOpen: true }),
```

Comment:
두 함수의 로직 형태가 다른 건 의도하신건가요??
```suggestion
  onSuccess: (text: string) => set(() => ({ isSuccess: true, text, isOpen: true })),
  onFailure: (text: string) => set(() => ({ isSuccess: false, text, isOpen: true })),
```

Reply:
저 같은 경우 state를 사용한다면 콜백함수를 사용, 그렇지 않다면 해당 값을 직접 선언하고 있습니다.

이런 의미를 코드로 직관적으로 볼 수 있기 때문에 다음과 같은 이유로 작성하고 있습니다.